src/components/ContactUsForm/index.jsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * ContactUsForm consists of validation and elements of contact form, imports Netlifyform for post data.
+ * The base code that is used in both netlifyForm and index are implemented from https://github.com/jon-fm/react-ssg-netlify-forms-demo
+ */
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import NetlifyForm from "../NetlifyForm";
+import { Col, Row } from "antd";
+import { RiSendPlaneFill } from "react-icons/ri";
+import "./index.css";
+
+const ContactUsForm = () => {
+  const [t] = useTranslation();
+  const [submissionMessage, setSubmissionMessage] = useState("");
+  // processing state used in disabling button when inputs are empty, changing when they are not.
+  const [processing, setProcessing] = useState(false);
+  // beforeSubmit is for validations and disabling submit button
+  const beforeSubmit = async () => {
+    if (
+      contactFormValues.name.length > 0 &&
+      contactFormValues.email.length > 0
+    ) {
+      setProcessing(true);
+      // Wait 2 seconds to simulate async delay
+      await new Promise((resolve) => setTimeout(resolve, 2000));
+      return true;
+    } else {
+      return false;
+    }
+  };
+  // TODO: get submission message to translate files.
+  // submisson message pops out when form is submitted
+  // afterSubmit is for submission message and enable submit button
+  const afterSubmit = () => {
+    setSubmissionMessage("Thanks for messaging us!");
+    setProcessing(false);
+  };
+  // handles value changes of contact form
+  const handleChange = (e) =>
+    setContactFormValues({
+      ...contactFormValues,
+      [e.target.name]: e.target.value,
+    });
+  const [contactFormValues, setContactFormValues] = useState({
+    name: "",
+    email: "",
+    message: "",
+  });
+  return (
+    <div className="contactUsForm">
+      <NetlifyForm
```

Comment:
i think if you delete this whole NetlifyForm thing and just add data-netlify=true to the form (https://docs.netlify.com/forms/setup/#html-forms) then it will refresh the page but also submit the data. did you try that?

because that would reduce submitting to netlify forms to one line rather than having this complicated component/onsubmit thing (see my other comment about copying a random person's code)